src/models/document/document-content-import.ts

```diff
@@ -91,10 +91,14 @@ export function migrateSnapshot(snapshot: IDocumentImportSnapshot): any {
   for (const id of tiles) {
     const tile = docContent.tileMap.get(id);
     if (tile && tile.title && getTileContentInfo(tile.content.type)?.useContentTitle) {
-      const sharedModel = docContent.getFirstSharedModelByType(SharedDataSet, tile.id);
-      if (sharedModel && isSharedDataSet(sharedModel)) {
-        sharedModel.dataSet.setName(tile.title);
-        tile.setTitle(undefined);
+      // Look for a SharedModel that can hold the title
+      for (const sm of Object.values(docContent.getSharedModelsUsedByTiles([id]))) {
+        if (getSharedModelInfoByType(sm.sharedModel.type)?.hasName) {
+          sm.sharedModel.setName(tile.title);
+          tile.setTitle(undefined);
+          console.log("Set title onto", sm.sharedModel);
```

Comment:
If this is useful to keep, it probably should be behind a debug flag. Or perhaps it is fine to always print it because it should be rare after most of the tiles are migrated. In either case it needs a eslint comment so it doesn't cause a eslint warning.

Reply:
looks like you got it already 👍